src/LLVM/Typed.hs

```diff
@@ -84,15 +86,19 @@ instance Typed C.Constant where
 getElementPtrType :: Type -> [C.Constant] -> Type
 getElementPtrType ty cons = ptr i8 -- XXX
 
+getElementType :: Type -> Type
```

Comment:
I agree, At first I thought that `ArrayType` is valid for `GetElementPtr` but the correct way for those is to point to a global constant variable